R/translate-sql-string.R

```diff
@@ -5,19 +5,17 @@
 #' @rdname sql_variant
 sql_substr <- function(f = "SUBSTR") {
   function(x, start, stop) {
-    start <- max(check_integer(start, "start"), 1L)
-    stop <- max(check_integer(stop, "stop"), 1L)
+    start <- max(check_integer(start), 1L)
+    stop <- max(check_integer(stop), 1L)
     length <- max(stop - start + 1L, 0L)
 
     sql_call2(f, x, start, length)
   }
 }
 
-check_integer <- function(x, arg) {
-  if (length(x) != 1 || !is.numeric(x)) {
-    cli_abort("{.arg {arg}} must be a single number")
-  }
-  as.integer(x)
+check_integer <- function(x, arg = caller_arg(x), call = caller_env()) {
```

Comment:
I think you should give this a different prefix since it also does coercion (which we're generally moving away from in `check_` functions)